src/contexts/SiteLaunchContext.tsx

```diff
@@ -0,0 +1,86 @@
+import { createContext, useContext, useEffect, useState } from "react"
+import { useParams } from "react-router-dom"
+
+import { useGetSiteLaunchStatus } from "hooks/siteDashboardHooks"
+
+import {
+  SiteLaunchDto,
+  SiteLaunchFrontEndStatus,
+  SiteLaunchStatusProps,
+  SITE_LAUNCH_TASKS_LENGTH,
+} from "types/siteLaunch"
+
+interface SiteLaunchContextProps {
+  siteLaunchStatusProps: SiteLaunchStatusProps
+  setSiteLaunchStatusProps: (
+    siteLaunchStatusProps: SiteLaunchStatusProps
+  ) => void
+}
+
+const SiteLaunchContext = createContext<SiteLaunchContextProps | null>(null)
+
+export const useSiteLaunchContext = (): SiteLaunchContextProps => {
+  const SiteLaunchContextData = useContext(SiteLaunchContext)
+  if (!SiteLaunchContextData)
+    throw new Error("useSiteLaunchContext must be used within an RoleProvider")
+  return SiteLaunchContextData
+}
+
+export const SiteLaunchProvider = ({
+  children,
+  initialSiteLaunchStatus = "NOT_LAUNCHED",
+  initialStepNumber = 0,
+}: {
+  children: React.ReactNode
+  initialSiteLaunchStatus?: SiteLaunchFrontEndStatus
+  initialStepNumber?: number
+}): JSX.Element => {
+  const { siteName } = useParams<{ siteName: string }>()
+
+  const [
+    siteLaunchStatusProps,
+    setSiteLaunchStatusProps,
+  ] = useState<SiteLaunchStatusProps>({
+    siteLaunchStatus: initialSiteLaunchStatus,
+    stepNumber: initialStepNumber,
+  })
```

Comment:
we don't have to do ti now but do remember that we'd need to feedback network failures to our users somehow

Reply:
Yup, agreed, this is only for the first cut!